docs/proposals/20221003-In-place-propagation-of-Kubernetes-objects-only-changes.md

```diff
@@ -0,0 +1,321 @@
+
+If Cluster API uses regular patches instead of SSA patches, but without being authoritative, Cluster API can
+be implemented in order to add new labels from higher level objects (e.g. a new label added to MachineSet is added to
+the corresponding Machine) and to enforce labels values from higher level objects. 
+
+But, as explained in [avoiding conflicts with other components](#avoiding-conflicts-with-other-components), using
+this approach there is no way to determine if label/annotation has been applied by Cluster API or by the user or another controllers,
+and thus automatic label/annotation deletion cannot be implemented.
+
+This approach is not considered ideal, because it is transferring the ownership of labels and annotations deletion
+to users or other controllers, and this is not considered a nice user experience.
+
+### To not use SSA for [in-place propagation](#in-place-propagation) and use status fields to track labels previously applied by CAPI
+
+If Cluster API uses regular patches instead of SSA patches, without being authoritative, it is possible to implement
+a DIY solution for tracking label ownership based on status fields or annotations.
+
+This approach is not considered ideal, because e.g. status field do not survive move/backup and restore, and tacking
+a step back, this is sort of re-implementing SSA or a subset of it.
+
+### Change more propagation rules
+
+While working on the set of changes proposed above a set optional changes to the existing propagation rules has been
+identified; however, considering that the more complex part of this proposal is implementing [in-place propagation](#in-place-propagation),
+it was decided to implement only the few, most critical changes to propagation rules. 
+
+Nevertheless we are documenting optional changes dropped from the scope of this iteration for future reference.
+
+![Figure 3](./images/in-place-propagation/optional-changes.png)
+
+Optional changes:
+
+- 4b: Simplify MachineDeployment to MachineSet label propagation
+  Leveraging on changed introduced 4, it is possible to simplify MachineDeployment to MachineSet label propagation,
+  which currently mimics Deployment to ReplicaSet label propagation. The backside of this chance is that it wouldn't be
+  possible anymore to have different labels/annotations on MachineDeployment & MachineSet.
+
+- 5a and 5b: Propagate ClusterClass and Cluster.topology to templates
+  This changes make ClusterClass and Cluster.topology labels/annotation to be propagated to templates as well.
+  Please note that this change requires further discussions, because
+  - Contract with providers should be extended to add optional metadata fields where necessary
+  - It should be defined how to detect if a template for a specif provider has the optional metadata fields,
+    and this is tricky because Cluster API doesn't have detailed knowledge of provider's types.
```

Comment:
If you want, let's also add:
- InfrastructureMachineTemplates are immutable in a lot of providers, so we have to discuss how/if we should be able to mutate the InfrastructureMachineTemplates.spec.template.metadata.